011-kafka-canary.md

```diff
@@ -0,0 +1,123 @@
+# Strimzi Canary
+
+Implement a Kafka canary tool which will act as an indicator of whether Kafka clusters are operating correctly.
+This would be achieved by creating a canary topic and periodically producing and consuming events on the topic. 
+
+## Current situation
+
+Currently Strimzi does not have any canary style feature.
+It does have health checks for the Kafka cluster which can be replaced/augmented with the canary.
+
+## Motivation
+
+The canary would provide the following in indication that the Kafka cluster is operating as expected from a users perspective. I.e. messages can successfully be produced and consumed. 
+Some would be exported to reflect the canary's activity. Initial metrics would included:
+  * records-produced-total
+  * records-consumed-total
+  * produce-error
+  * consume-error
+  * latency
+
+In future the canary can be used by the Kafka Roller to improve health checks. 
+However this is beyond the scope of the current proposed work.
+
+## Proposal
+
+The canary would be run as a separate pod alongside Strimzi operator. 
+Once the canary topic is created messages will be produced and consumed from it. 
```

Comment:
Should we cover here the configuration of the canary topic? In general, we should probably consider using smaller segment size and retention to make sure this does not unnecessarily fill the disk space. Not sure it needs to be in the proposal, probably not. Just a thought which should definitely not get forgotten in the impl.

Reply:
I added a sentence in the "Topic Configuration" section but without numbers about segment size and retention (in bytes). It will come with development and more testing.